### PR TITLE
Adds support for walls

### PR DESCRIPTION
**Pull request by hankedan000**
Adds logic to the game for supporting walls.
 * Parses '_obstacle' of map data files
 * Spawns walls in track
   - supports both "full height" and "crouch" type walls
 * Quiets song audio when player enters wall
 * Introduced collision layers into the project
 * Shifted cubes slightly up for better playability (can be made dynamic based on player height in the future)

### CODE REVIEW

File: game/BeepSaber_Game.gd [hunk 1]
```diff
@@ -132,36 +141,86 @@ func _spawn_cube(note, current_beat):
...
 
 	cube._cube_mesh_orientation.rotation.z = rotation_z;
 	if note._cutDirection==8:
 		cube._cube_mesh_orientation.rotation.y = deg2rad(180);
 
 	cube._note = note;
 
+# constants used to interpret the '_type' field in map obstacles
+const WALL_TYPE_FULL_HEIGHT = 0;
+const WALL_TYPE_CROUTCH = 1;
```
Comment (hankedan000) [inline game/BeepSaber_Game.gd]: crouch is spelled wrong

**Comment by NeoSpark314** [comment]
Nice! That looks great. Did you test it on Quest? I will give it a closer look later but from what I see in the code it looks good to be merged and we can solve potential issues later.

**Comment by NeoSpark314** [comment]
Reason I'm asking for quest is that transparency is very expensive to render on Quest and the game was already running close to the performance limit on Quest 1; so I fear that transparent walls might result in framerate/performance issues...

**Comment by hankedan000** [comment]
Thanks! It's a fun game to play. I'm happy to contribute.
No, I only have a quest 2. It runs well on that, but obviously that has more power than quest 1. I ended up making the walls an opaque marble material.

**Comment by NeoSpark314** [comment]
Finally merged it to master, sorry for the delay and thanks again for this nice contribution. Do you have a suggestion of a song I could use for testing just to see how the performance on a Quest 1 is?

**Comment by hankedan000** [comment]
The included demo song has walls in it, but it's not overly strenuous on wall generation/movement.
I always enjoy playing Shutup And Dance by Walk The Moon. It has a fun intro with walls.
https://bsaber.com/songs/2144/
Back In Black by ACDC has a fun ending.
https://bsaber.com/songs/9d7/
After you asked this question, it made me look around for something a little more testing. I found Eizo's Family. It's a fun "dodger" type map on Hard. Expert has some notes thrown in. The two maps play well with almost no noticeable lag on the Quest 2. It has however, made me realize a few bugs in the wall generation though. I also noticed that the zero thickness walls suffer from texture overlap artifacts (which I new about, but didn't think it was a huge deal breaker).
https://bsaber.com/songs/72f/

**Comment by NeoSpark314** [comment]
Thanks for the suggestions! So far performance with walls still looks good on Quest 1.

**Pull request merged**